8.7.0-Alpha3 Release Notes (#4759)
camunda/camunda-docs: docs/reference/release-notes/870.md

| ---------------------- | ---------------------------- | ------------ | ------------ | ------------ |
| 11 February 2025       | 11 August 2026               | -            | -            | -            |

## 8.7.0-alpha3

| Release date    | Changelog(s)                                                                                                                                                                               | Blog                                                                                |
| :-------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------- |
| 14 January 2025 | <ul><li>[ Camunda 8 core ](https://github.com/camunda/camunda/releases/tag/8.7.0-alpha3)</li><li>[ Connectors ](https://github.com/camunda/connectors/releases/tag/8.7.0-alpha3)</li></ul> | [Release blog](https://camunda.com/blog/2025/1/camunda-alpha-release-january-2025/) |

<!-- https://github.com/camunda/product-hub/issues/2546 -->

> ### Ad-hoc subprocess BPMN pattern support <span class="badge badge--long" title="This feature affects SaaS">SaaS</span><span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span><span class="badge badge--medium" title="This feature affects Modeler">Modeler</span>

Reviewer: This feature did not make it into the Modelers yet. You can deploy a hand-crafted XML to the engine, but you cannot model it in a diagram. I'd suggest to note it in the section explicitly, and we can announce the full support in the next alpha release notes as we are aiming to have modeling support by then.

Author: Thanks @barmac - I added in a note - is this accurate?

> This release only supports manual deployment of the element to the engine via XML. Modeler support for this element is planned for the next alpha release.

Reviewer: I assume this is outdated given https://github.com/camunda/camunda-docs/pull/4759#issuecomment-2582293910

<!-- https://github.com/camunda/product-hub/issues/2374 -->

### Camunda cluster update API <span class="badge badge--long" title="This feature affects SaaS">SaaS</span><span class="badge badge--medium" title="This feature affects APIs">API</span>

You can now use the Camunda 8 SaaS [Administration API](/apis-tools/administration-api/administration-api-reference.md) to update a cluster to a new generation.

- This allows you to trigger automated simultaneous updates for multiple clusters via the API.
- Send a PUT request to the Administration API `Update cluster` endpoint.

### Connectors <span class="badge badge--long" title="This feature affects SaaS">SaaS</span><span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span><span class="badge badge--medium" title="This feature affects Connectors">Connectors</span>

New Connectors and enhancements are included in this release.

<!-- https://github.com/camunda/product-hub/issues/2552 -->

#### AWS S3 Connector

Use the new outbound Amazon S3 Connector to interact with [Amazon Simple Storage Service (Amazon S3)](https://aws.amazon.com/S3/) from your BPMN process.

> This Connector supports the following actions:

Reviewer 2: Nit-pick: Do Connectors have actions, operations, or both? And what is the difference? Can we standardize on this.

Author: I was checking with this, and yes Operations is the way to go, I've changed this - good spot! 🚀

- Upload Document: Upload a document to an AWS S3 bucket.
- Download Document: Download a document from an AWS S3 bucket.
- Delete Document: Delete a document from an AWS S3 bucket.

<!-- https://github.com/camunda/product-hub/issues/2399 -->

#### Box Connector

Use the new outbound Box connector to interact with [Box.com](https://www.box.com/) account content from your BPMN process.

This Connector supports the following operations:

- Download File: Download files into the Camunda document store.
- Upload File: Upload files from Camunda to Box.
- Delete File: Delete file items in Box.
- Move File: Move files between folders in Box.
- Create Folder: Create new folders in your Box account.
- Delete Folder: Delete folders from your Box account.

> - Search: Search file items using the Box search API.

Reviewer 2: This is a little nitpicky, but I didn't like how this list looks on the preview site. I have two suggestions you can explore:

Bold the text to the left of the colon
> **Download File**: Download files into the Camunda document store.
(Following our style guide, I think this is actually needs to be lowercase "file" too.)

Or, remove the text to the left of the colon entirely.
> Download files into the Camunda document store.

This should be applied to the other Connector as well, but it jumped out at me in this longer list.

Author: Agreed, changed 👍

<!-- https://github.com/camunda/product-hub/issues/2555 -->
<!-- https://github.com/camunda/product-hub/issues/2592 -->
<!-- https://github.com/camunda/connectors/pull/3751 -->
<!-- https://github.com/camunda/connectors/pull/3746 -->

### Document handling <span class="badge badge--long" title="This feature affects SaaS">SaaS</span><span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span><span class="badge badge--medium" title="This feature affects APIs">API</span><span class="badge badge--medium" title="This feature affects Modeler">Modeler</span>

New features are available as part of the enhanced document handling being delivered with the 8.7 release.

> - The Document store API now supports uploading files in batches.

Reviewer 2: "The Document store API" casing looks a bit odd.

Author: Agreed, changed 👍

Web Modeler now supports native integration between a process application and a branch of a GitLab repository.

Non-technical users in orgs running GitLab can now easily access the files in their source of truth, collaborate cross-platform with Desktop Modeler users, and contribute changes to a feature branch that can be easily merged and deployed.

To learn more about this feature, see [Git sync](/components/modeler/web-modeler/git-sync.md).

<!-- https://github.com/camunda/product-hub/issues/2473 -->

### Optimize Amazon OpenSearch support <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span><span class="badge badge--medium" title="This feature affects Optimize">Optimize</span>

Camunda 8 Self-Managed now fully supports the use of Amazon OpenSearch with Optimize.

<!-- https://github.com/camunda/product-hub/issues/2502 -->

### Process Instance Migration <span class="badge badge--long" title="This feature affects SaaS">SaaS</span><span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span><span class="badge badge--medium" title="This feature affects Zeebe">Zeebe</span>

> Enhanced Process Instance Migration now supports additional BPMN elements, enabling smoother transitions for your workflows and empowering operations teams to seamlessly migrate running process instances to updated process definitions without workarounds or manual intervention.

Reviewer 2: ```suggestion
Enhanced process instance migration now supports additional BPMN elements, enabling smoother transitions for your workflows and empowering operations teams to seamlessly migrate running process instances to updated process definitions without workarounds or manual intervention.
```

This enhancement streamlines operations and ensures smoother incident handling, improving time to unblock process execution.

<!-- https://github.com/camunda/product-hub/issues/2073 -->

### Unified deployment experience from Web Modeler <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span><span class="badge badge--medium" title="This feature affects Modeler">Modeler</span>

The deployment experience is further simplified for Enterprise customers running Web Modeler Self-Managed.

The predefined development stage cluster for the process application is also used when launching Play. This feature is already available in SaaS, and is now also available for Self-Managed as part of unifying the deployment experience.

<!-- https://github.com/camunda/product-hub/issues/2126 -->

### Zeebe user tasks modeling migration support <span class="badge badge--long" title="This feature affects SaaS">SaaS</span><span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span><span class="badge badge--medium" title="This feature affects Modeler">Modeler</span>

> User task events with the Zeebe user task implementation are renamed to "Camunda user tasks" and set as the default Modeler type.

Reviewer: ```suggestion
User task implementation type "Zeebe user task" is renamed to "Camunda user task", and set as the default implementation type.
```

Reviewer: Old:

![image](https://github.com/user-attachments/assets/28057b11-f9cd-427f-818a-75502e694308)

New:
![image](https://github.com/user-attachments/assets/c1c87a52-e346-4f33-a711-4f047715ae13)

:::note

> Job worker-based user tasks are now deprecated, with migration support provided to help you transition to the new type.

Author: Thanks @barmac, would this be better as a note:

> Note: As Job-worker user tasks will be deprecated in Camunda 8.9, Camunda recommends you start using Camunda User Tasks (formerly known as Zeebe User Task) in your process definitions. To learn more, see [Announcements](https://docs.camunda.io/docs/next/reference/announcements/announcements-870/#deprecated-job-based-user-tasks-querying).

> Job worker-based user tasks are now deprecated, with migration support provided to help you transition to the new type.

Reviewer: 100% precise would be: 
> Note: As Job-worker user tasks managed by Camunda will be deprecated...

> Job worker-based user tasks are now deprecated, with migration support provided to help you transition to the new type.

Author: Thanks @barmac I've changed this note as per your feedback 🚀

## 8.7.0-alpha3

| Release date    | Changelog(s)                                                                                                                                                                               | Blog                                                                                |
| :-------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------- |
| 14 January 2025 | <ul><li>[ Camunda 8 core ](https://github.com/camunda/camunda/releases/tag/8.7.0-alpha3)</li><li>[ Connectors ](https://github.com/camunda/connectors/releases/tag/8.7.0-alpha3)</li></ul> | [Release blog](https://camunda.com/blog/2025/1/camunda-alpha-release-january-2025/) |

<!-- https://github.com/camunda/product-hub/issues/2473 -->

### Amazon OpenSearch Optimize support <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span><span class="badge badge--medium" title="This feature affects Optimize">Optimize</span>

> Camunda 8 Self-Managed now fully supports the use of Amazon OpenSearch with Optimize.

Reviewer 2: Hmm... following the trail from the epic I found https://github.com/camunda/camunda-docs/pull/4750/files where I can see the docs were just made a bit more generic. 

I can think of a couple ways we may want to address this:
* Relate this change back to Elasticsearch. Something that hints at why the parameter names used for OpenSearch may still retain the Elasticsearch name.
* Summarize the docs changes associated with this epic. Where the docs previously explained configuration exclusively for Elasticsearch, you'll find a generic reference to "the database" and can use either Elasticsearch or OpenSearch. 

Again, this can be a follow-up item if you want to sync with the implementation team. It just stood out to me that there was no link to the docs.

> Camunda 8 Self-Managed now fully supports the use of Amazon OpenSearch with Optimize.

Author: I'll follow up with a PR for this 👍
